corehq/apps/hqcase/utils.py

```diff
@@ -44,7 +44,8 @@ def submit_case_blocks(case_blocks, domain, username="system", user_id="",
         'time': now,
         'uid': form_id,
         'username': username,
-        'user_id': user_id,
+        'user_id': user_id or "",
```

Comment:
I changed the default from `""` to `None` so that if I call `submit_case_blocks(..., user_id=user_id, ...)` and `user_id` is `None`, the form will still be rendered with the value `""` for `user_id`.

From an interface perspective, this change also makes the function's signature "more correct". The right way to tell a function "I don't have a `user_id`" is to pass in `user_id=None`; `user_id=""` means "I have a `user_id`, and it's `""`"